src/imp/security_framework.rs

```diff
@@ -78,34 +74,14 @@ pub struct Pkcs12 {
 
 impl Pkcs12 {
     pub fn from_der(buf: &[u8], pass: &str) -> Result<Pkcs12, Error> {
-        let dir = match TempDir::new("native_tls") {
-            Ok(dir) => dir,
-            // Gotta throw away the real error :(
-            Err(_) => return Err(Error(base::Error::from(errSecIO))),
-        };
-
-        let keychain = try!(keychain::CreateOptions::new()
-            .password(pass) // FIXME maybe generate a secure random password here?
-            .create(dir.path().join("keychain")));
-
-        let mut import = try!(Pkcs12ImportOptions::new()
-            .passphrase(pass)
-            .keychain(keychain)
-            .import(buf));
-        let import = import.pop().unwrap();
-
-        // The identity's cert shows up in the chain, so filter it out to avoid sending twice
-        // FIXME should probably use CFEquals here
-        let identity_cert = try!(import.identity.certificate()).to_der();
-
-        Ok(Pkcs12 {
-            identity: import.identity,
-            // FIXME possibly use the chain from the trust result instead?
-            chain: import.cert_chain
-                .into_iter()
-                .filter(|c| c.to_der() != identity_cert)
-                .collect(),
-        })
+        let mut options = Pkcs12ImportOptions::new();
+        options.passphrase(pass);
+
+        let mut imported_identities = options.import(buf)?;
+        imported_identities.truncate(1);
+        let imported_identity = imported_identities.pop().unwrap();
+        
+        Ok(Pkcs12{identity: imported_identity.identity, chain: imported_identity.cert_chain})
```

Comment:
I believe we'll want to continue filtering the leaf certificate out of the chain.